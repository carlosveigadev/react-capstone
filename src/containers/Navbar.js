import { Link } from 'react-router-dom';

function Navbar() {
  return (
    <>
      <h2>Pokemon Pokedex</h2>
      <Link to="/">Home </Link>
      <Link to="/about">About </Link>
    </>
  );
}

export default Navbar;
