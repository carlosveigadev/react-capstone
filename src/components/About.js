import Navbar from '../containers/Navbar';
import style from '../styles/About.module.css';

const About = () => (
  <>
    <Navbar />
    <h1 className={style.center}>
      This is a project made using react-redux and Pokemon API.
    </h1>
  </>
);

export default About;
