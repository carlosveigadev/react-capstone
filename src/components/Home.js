import Navbar from '../containers/Navbar';
// import Category from '../containers/Category';
import PokemonList from '../containers/PokemonList';

const Home = () => (
  <>
    <Navbar />
    <h1>Hi!</h1>
    <PokemonList />
  </>
);

export default Home;
