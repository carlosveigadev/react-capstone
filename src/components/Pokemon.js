import PropTypes from 'prop-types';
import { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { GiBodyHeight, GiWeight } from 'react-icons/gi';
import { HiOutlineIdentification } from 'react-icons/hi';
import Navbar from '../containers/Navbar';
import style from '../styles/Pokemon.module.css';
import { fetchPokeData } from '../api-requests/request';

const Pokemon = ({ location }) => {
  const [data, setData] = useState(null);

  let name;
  let pokeUrl;

  if (location.state === undefined) {
    const { pokeName } = useParams();
    name = pokeName;
    pokeUrl = `https://pokeapi.co/api/v2/pokemon/${pokeName}/`;
  } else {
    name = location.state.poke;
    pokeUrl = location.state.url;
  }

  useEffect(async () => {
    const newData = await fetchPokeData(pokeUrl);
    setData(newData);
  }, []);

  if (data) {
    return (
      <span>
        <Navbar />
        <div className={style.flex}>
          <div className={style.mainDiv}>

            <h1>{name}</h1>
            <img src={data.sprites.front_default} alt={name} />

            <div className={style.flex}>
              <GiBodyHeight className={style.icon} />
              <div className={style.data}>
                {`${data.height * 10}cm`}
              </div>
            </div>

            <div className={style.flex}>
              <GiWeight className={style.icon} />
              <div className={style.data}>
                {`${data.weight / 10}kg`}
              </div>
            </div>

            <div className={style.flex}>
              <HiOutlineIdentification className={style.icon} />
              <div className={style.data}>
                {`${data.id}`}
              </div>
            </div>

          </div>
        </div>
      </span>
    );
  }
  return (
    <p className={style.flex}>Fetching Data...</p>
  );
};

Pokemon.propTypes = {
  location: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default Pokemon;
