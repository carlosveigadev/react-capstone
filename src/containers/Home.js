import Navbar from './Navbar';
import PokemonList from '../components/PokemonList';
import style from '../styles/Home.module.css';

const Home = () => (
  <>
    <Navbar />
    <h1 className={style.title}>
      Hi, here you can select to show all the pok√©mons or you can select a specific type!
    </h1>
    <PokemonList />
  </>
);

export default Home;
