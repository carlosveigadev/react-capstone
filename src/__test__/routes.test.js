import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { createStore, combineReducers } from 'redux';
import { Provider } from 'react-redux';
import Navbar from '../containers/Navbar';
import About from '../components/About';
import Pokemon from '../components/Pokemon';
import Home from '../components/Home';
import filterReducer from '../reducers/filter';
import pokeReducer from '../reducers/pokemon';
import catReducer from '../reducers/category';

describe('Check if routes work ', () => {
  it('renders Navbar page correctly in the proper route.', () => {
    const navbar = renderer
      .create(
        <BrowserRouter>
          <Navbar />
        </BrowserRouter>,

      )
      .toJSON();
    expect(navbar).toMatchSnapshot();
  });

  it('renders About page correctly in the proper route.', () => {
    const about = renderer
      .create(
        <BrowserRouter>
          <About />
        </BrowserRouter>,

      )
      .toJSON();
    expect(about).toMatchSnapshot();
  });

  it('renders Pokemon page correctly in the proper route.', () => {
    const props = { state: { poke: 'test', url: 'testurl' } };
    const poke = renderer
      .create(
        <BrowserRouter>
          <Pokemon location={props} />
        </BrowserRouter>,
      )
      .toJSON();
    expect(poke).toMatchSnapshot();
  });

  it('renders Home page correctly in the proper route.', () => {
    const rootReducer = combineReducers({
      pokes: pokeReducer,
      filter: filterReducer,
      categoryPokemon: catReducer,
    });
    const store = createStore(rootReducer);
    const home = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Home />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(home).toMatchSnapshot();
  });
});
