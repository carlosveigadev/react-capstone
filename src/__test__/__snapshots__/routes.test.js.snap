// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if routes work  renders About page correctly in the proper route. 1`] = `
Array [
  <h2
    className="title"
  >
    Pokemon Pokédex
  </h2>,
  <div
    className="navbarItems"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Home 
    </a>
    <a
      href="/about"
      onClick={[Function]}
    >
      About 
    </a>
  </div>,
  <h1
    className="center"
  >
    This is a project made using react-redux and Pokemon API.
  </h1>,
]
`;

exports[`Check if routes work  renders Home page correctly in the proper route. 1`] = `
Array [
  <h2
    className="title"
  >
    Pokemon Pokédex
  </h2>,
  <div
    className="navbarItems"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Home 
    </a>
    <a
      href="/about"
      onClick={[Function]}
    >
      About 
    </a>
  </div>,
  <h1
    className="title"
  >
    Hi, here you can select to show all the pokémons or you can select a specific type!
  </h1>,
  <div>
    <div
      className="categoryDiv"
    >
      <div
        className="customSelect"
      >
        <select
          onChange={[Function]}
        >
          <option
            value="All"
          >
            All
          </option>
          <option
            value="Normal"
          >
            Normal
          </option>
          <option
            value="Fighting"
          >
            Fighting
          </option>
          <option
            value="Flying"
          >
            Flying
          </option>
          <option
            value="Poison"
          >
            Poison
          </option>
          <option
            value="Ground"
          >
            Ground
          </option>
          <option
            value="Rock"
          >
            Rock
          </option>
          <option
            value="Bug"
          >
            Bug
          </option>
          <option
            value="Ghost"
          >
            Ghost
          </option>
          <option
            value="Steel"
          >
            Steel
          </option>
          <option
            value="Fire"
          >
            Fire
          </option>
          <option
            value="Water"
          >
            Water
          </option>
          <option
            value="Grass"
          >
            Grass
          </option>
          <option
            value="Electric"
          >
            Electric
          </option>
          <option
            value="Psychic"
          >
            Psychic
          </option>
          <option
            value="Ice"
          >
            Ice
          </option>
          <option
            value="Dragon"
          >
            Dragon
          </option>
          <option
            value="Dark"
          >
            Dark
          </option>
          <option
            value="Fairy"
          >
            Fairy
          </option>
          <option
            value="Unknown"
          >
            Unknown
          </option>
          <option
            value="Shadow"
          >
            Shadow
          </option>
        </select>
      </div>
    </div>
    <div
      className="pokemonDiv"
    >
      <div />
    </div>
  </div>,
]
`;

exports[`Check if routes work  renders Navbar page correctly in the proper route. 1`] = `
Array [
  <h2
    className="title"
  >
    Pokemon Pokédex
  </h2>,
  <div
    className="navbarItems"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Home 
    </a>
    <a
      href="/about"
      onClick={[Function]}
    >
      About 
    </a>
  </div>,
]
`;

exports[`Check if routes work  renders Pokemon page correctly in the proper route. 1`] = `
<p
  className="flex"
>
  Fetching Data...
</p>
`;
